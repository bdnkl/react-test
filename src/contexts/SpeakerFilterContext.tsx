import React, {createContext} from "react";
import useSpeakerFilter from "../hooks/useSpeakerFilter";

interface SpeakerFilterContextValue {
  showSessions: boolean,
  setShowSessions: (arg: boolean) => void,
  eventYear: string,
  setEventYear: (arg: string) => void,
  searchQuery: string,
  setSearchQuery: (arg: string) => void,
  EVENT_YEARS: Array<string>
}

const SpeakerFilterContext = createContext<SpeakerFilterContextValue | undefined>(undefined);

interface SpeakerFilterProviderProps {
  children: React.ReactNode,
  startingShowSessions?: boolean,
  startingEventYear?: string
}

export function SpeakerFilterProvider({children, startingShowSessions = false, startingEventYear = "2019",}
                                        : SpeakerFilterProviderProps) {
  const {
    showSessions,
    setShowSessions,
    eventYear,
    setEventYear,
    searchQuery,
    setSearchQuery,
    EVENT_YEARS,
  } = useSpeakerFilter(startingShowSessions, startingEventYear);

  return (
    <SpeakerFilterContext.Provider
      value={{
        showSessions,
        setShowSessions,
        eventYear,
        setEventYear,
        searchQuery,
        setSearchQuery,
        EVENT_YEARS,
      }}
    >
      {children}
    </SpeakerFilterContext.Provider>
  );
}

export function useSpeakerFilterContext() {
  const context = React.useContext(SpeakerFilterContext)

  if (context === undefined) {
    throw new Error('useSpeakerContext should be used within an SpeakerFilterProvider.')
  }

  return context
}
