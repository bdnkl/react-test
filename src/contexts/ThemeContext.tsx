import React, {createContext} from "react";

import useTheme, {ThemeType} from "../hooks/useTheme";

interface ThemeContextValue {
  theme: ThemeType,
  setTheme: (arg: ThemeType) => void
}

export const ThemeContext = createContext<ThemeContextValue | undefined>(undefined);

export function ThemeProvider({children, startingTheme}) {
  const {theme, setTheme} = useTheme(startingTheme);

  function handleSetTheme(newTheme: ThemeType) {
    setTheme(newTheme);
  }

  return (
    <ThemeContext.Provider value={{theme, setTheme: handleSetTheme}}>
      {children}
    </ThemeContext.Provider>
  );
}

export function useThemeContext() {
  const context = React.useContext(ThemeContext)


  if (context === undefined) {
    throw new Error('useThemeContext should be used within an ThemeProvider.')
  }

  return context
}
