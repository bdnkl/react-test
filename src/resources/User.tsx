export const userDefault = "Ronald";
export type UserType = string;
