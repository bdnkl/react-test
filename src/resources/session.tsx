type RoomType = {
    name: string;
    capacity: number;
}

export type SessionType = {
    id: string;
    title: string;
    eventYear: string;
    room: RoomType;
}
