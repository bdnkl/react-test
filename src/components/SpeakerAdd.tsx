import React from "react";
import withAuth from "./withAuth";
import { UserType } from "../resources/User";
import { speakerDefault, SpeakerType } from "../resources/Speaker";

interface SpeakerAddInterface {
  eventYear: string;
  insertRecord: (speaker: SpeakerType) => void;
  loggedInUser: UserType;
}

export function SpeakerAdd({
  eventYear,
  insertRecord,
  loggedInUser,
}: SpeakerAddInterface) {
  if (!loggedInUser || loggedInUser.length === 0) return null;

  return (
    <a href="#" className="addSes">
      <i
        onClick={(e) => {
          e.preventDefault();
          const firstLast = window.prompt("Enter first and last name:", "");
          const firstLastArray = firstLast.split(" ");
          insertRecord({
            ...speakerDefault,
            id: "99999",
            first: firstLastArray[0],
            last: firstLastArray[1],
            bio: "Bio not entered yet",
            sessions: [
              {
                id: "88888",
                title: `New Session For ${firstLastArray[0]}`,
                room: {
                  name: "Main Ball Room",
                },
                eventYear,
              },
            ],
          });
        }}
      >
        +
      </i>
    </a>
  );
}

export default withAuth(SpeakerAdd);
