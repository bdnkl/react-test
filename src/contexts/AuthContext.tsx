import React, {createContext, useState} from "react";
import {UserType} from "../resources/User";

interface AuthContextValues {
  loggedInUser: UserType | null,
  setLoggedInUser: (user: UserType) => void
}

const AuthContext = createContext<undefined | AuthContextValues>(undefined);

export function AuthProvider({children, initialLoggedInUser}) {
  const [loggedInUser, setLoggedInUser] = useState<string>(initialLoggedInUser);

  function handleLoggedInUser(user: UserType) {
    setLoggedInUser(user);
  }

  return (
    <AuthContext.Provider value={{loggedInUser, setLoggedInUser: handleLoggedInUser}}>
      {children}
    </AuthContext.Provider>
  );
}

export default function useAuthContext() {
  const context = React.useContext(AuthContext)

  if (context === undefined){
    throw new Error('useAuthContext should be used within an AuthProvider.')
  }

  return context
}

