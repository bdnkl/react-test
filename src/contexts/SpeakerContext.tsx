import React, {createContext} from "react";
import {SpeakerType} from "../resources/Speaker";

export interface SpeakerContextValue {
  speaker: SpeakerType,
  updateRecord: (arg: SpeakerType, callback?: () => void) => void,
  insertRecord: (arg: SpeakerType, callback?: () => void) => void,
  deleteRecord: (arg: SpeakerType, callback?: () => void) => void,
}


const SpeakerContext = createContext<SpeakerContextValue | undefined>(undefined);

export function SpeakerProvider({children, speaker, updateRecord, insertRecord, deleteRecord}:
                                  SpeakerContextValue & { children: React.ReactElement }) {
  return (
    <SpeakerContext.Provider
      value={{speaker, updateRecord, insertRecord, deleteRecord}}
    >
      {children}
    </SpeakerContext.Provider>
  );
}

export function useSpeakerContext() {
  const context = React.useContext(SpeakerContext)

  if (context === undefined) {
    throw new Error('useSpeakerContext should be used within an SpeakerProvider.')
  }

  return context
}

