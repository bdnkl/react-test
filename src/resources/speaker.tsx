import {SessionType} from "./session";

export type SpeakerType = {
  id: string;
  first: string;
  last: string;
  company: string;
  bio: string;
  twitterHandle: string;
  favorite: boolean;
  sessions: Array<SessionType>;
}
