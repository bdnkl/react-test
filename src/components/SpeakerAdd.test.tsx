import React from "react";
import renderer from "react-test-renderer";
import {SpeakerType} from "../resources/Speaker";
import {SpeakerFilterProvider} from "../contexts/SpeakerFilterContext";
import {SpeakerAdd} from "./SpeakerAdd";

describe("Speaker Add Component", () => {
  // define speaker object
  // const speaker: SpeakerType = {
  //   id: "0",
  //   first: "Jane",
  //   last: "Doe",
  //   bio: "biography",
  //   company: "GitHub",
  //   favorite: true,
  //   twitterHandle: "janedoe",
  //   sessions: [
  //     {
  //       id: "0",
  //       title: "Janes session",
  //       eventYear: "2021",
  //       room: {name: "Room", capacity: 100},
  //     },
  //   ],
  // };

  it("renders correctly", () => {
    // create rendered tree of component
    const tree = renderer.create(
      <SpeakerFilterProvider
        startingShowSessions={true}
        startingEventYear={"2021"}
      >
        <SpeakerAdd
          eventYear="2019"
          insertRecord={(speaker: SpeakerType) => {
            console.log(speaker)
            /** TODO add a mockup for inserting records */
          }}
          loggedInUser="Jane"
        />
      </SpeakerFilterProvider>
    );

    // snapshot test
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
