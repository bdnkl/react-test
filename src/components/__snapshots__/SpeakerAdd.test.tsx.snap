// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speaker Add Component renders correctly 1`] = `
<a
  className="addSes"
  href="#"
>
  <i
    onClick={[Function]}
  >
    +
  </i>
</a>
`;
